File: src/main/java/com/ruiyu/outdoor/model/TrackPoint.java
package com.ruiyu.outdoor.model;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * GPS轨迹点实体
 */
@Data
@TableName("track_points")
@AllArgsConstructor
@NoArgsConstructor
public class TrackPoint {

    @TableId(type = IdType.AUTO)
    private Long id;

    /** 活动ID */
    private Long activityId;

    /** 纬度 */
    private Double latitude;

    /** 经度 */
    private Double longitude;

    /** 海拔(米) */
    private Double altitude;

    /** 速度(km/h) */
    private Double speed;

    /** 定位时间 */
    private LocalDateTime timestamp;

    /** 创建时间 */
    private LocalDateTime createdAt;
}

File: ios/OutdoorApp/Sources/Models/TrackPoint.swift
import Foundation

struct TrackPoint: Codable {
    let id: Int?
    let activityId: Int?
    let latitude: Double
    let longitude: Double
    let altitude: Double?
    let speed: Double?
    let timestamp: String?
    let createdAt: String?

    enum CodingKeys: String, CodingKey {
        case id, latitude, longitude, altitude, speed, timestamp
        case activityId = "activity_id"
        case createdAt = "created_at"
    }
}

struct TrackPointReq: Codable {
    let activityId: Int?
    let latitude: Double
    let longitude: Double
    let altitude: Double?
    let speed: Double?

    enum CodingKeys: String, CodingKey {
        case latitude, longitude, altitude, speed
        case activityId = "activity_id"
    }
}
